src/main/java/net/openhft/chronicle/wire/AbstractGeneratedMethodReader.java

```diff
@@ -243,7 +238,7 @@ public boolean readOne() {
                 context.rollbackOnClose();
         }
 
-        return ok || delegate().readOne();
```

Comment:
this should improve performance for when there is nothing to read

Reply:
Also this should improve performace if we process "unwanted" event, e.g. if a service processes an event that it's not interested in

Reply:
See above comment - we always return true if there was data to be read. Either the service processes it, or the defaultParser does

Reply:
Yes - this is correct. My point was that services now will process "unexpected" events faster, which is a very real-life scenario. In case e.g. we have a service that's everyone listens to (but each listener is interested in different events).